meson.build

```diff
@@ -0,0 +1,54 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: '2.0.0.dev0',
+    license: 'BSD-3',
+    meson_version: '>=0.64',
+    default_options: [
+        # TODO: investigate, does meson try to compile against debug Python
+        # when buildtype = debug, this seems to be causing problems on CI
+        # where provided Python is not compiled in debug mode
+        'buildtype=release',
+        # TODO: turn on werror when ready
+        #'werror=true',
+        # TODO: We are using POSIX functions(strdup in ujson), so we can't compile
+        # with strict C99 :(
+        #'c_std=c99'
+    ]
+)
+
+add_global_arguments('-DNPY_NO_DEPRECATED_API=0', language : 'c')
+add_global_arguments('-DNPY_NO_DEPRECATED_API=0', language : 'cpp')
+
+# This is a cython bug
```

Comment:
Guessing this only matters if the release type is debug?

Reply:
IIUC, asserts can still be hit in release mode if NDEBUG is not defined. It was at least for me.

Reply:
How do you see what flags are being passed to the compiler / linker? Surprised that a release build wouldn't already do this

Reply:
You can look at `build/build.ninja` or at `build/compile_commands.json`. Or use `meson introspect` in the build dir for various fancier options.

Reply:
Is the build directory persisted after running `python -m build`? Couldn't find either of these files or a directory to run meson introspect in

Reply:
Ooooooooh, I see now. By default, meson will set NDEBUG if buildtype is release. However, meson-python overwrites this because they are using a custom debug and optimization level. (Since it uses O2, meson doesn't recognize it as release mode). So NDEBUG doesn't get set. This seems like a meson-python bug.

Reference: https://mesonbuild.com/Builtin-options.html. Pay attention to b_ndebug, and the table with buildtype, debug, and optimization level.

Reply:
The build directory can be configured by passing the `builddir=path/to/builddir` option as a backend option in `pip` / `build`, so that meson-python uses that instead of a temporary directory.

This also allows you to configure meson yourself with custom options and then have meson-python reuse that as-is, IIRC.